Modsenfy.DataAccessLayer/Repositories/TrackRepository.cs

```diff
@@ -12,111 +12,118 @@
 
 namespace Modsenfy.DataAccessLayer.Repositories
 {
-    public class TrackRepository : ITrackRepository
-    {
-        private readonly DatabaseContext _databaseContext;
+	public class TrackRepository : ITrackRepository
+	{
+		private readonly DatabaseContext _databaseContext;
 
-        public TrackRepository(DatabaseContext databaseContext)
-        {
-            _databaseContext = databaseContext;
-        }
+		public TrackRepository(DatabaseContext databaseContext)
+		{
+			_databaseContext = databaseContext;
+		}
 
-        public async Task Create(Track entity)
-        {
-            await _databaseContext.AddAsync(entity);
-        }
+		public async Task Create(Track entity)
+		{
+			await _databaseContext.Tracks.AddAsync(entity);
+		}
 
-        public void Delete(Track entity)
-        {
-            _databaseContext.Remove(entity);
-            return;
-        }
+		public async Task<Track> CreateAndGet(Track entity)
+		{
+			var trackEntry = await _databaseContext.Tracks.AddAsync(entity);
+            await SaveChanges();
```

Comment:
It is better to remove savechanges from every CRUD in repository and move it to separate method(call this method from services)